truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/nodes/Node.java

```diff
@@ -517,6 +517,17 @@ public String getLanguage() {
         return "";
     }
 
+    /**
+     * Reports the execution count of a loop that is a child of this node. The optimization
+     * heuristics can use the loop count to guide compilation and inlining.
+     *
+     * @count count the number iterations to report to the runtime system
+     */
+    public final void reportLoopCount(int iterations) {
```

Comment:
From my naive, external point of view this looks very something similar to Profile. Shouldn't the API look like the profiles one?

I know that Profile doesn't need any special support from the runtime and apparently loop count reporting needs it. That means internally these two things are probably very different, but from user point of view I don't see much difference in loop counting and branch probability profiling.